index.js

```diff
@@ -0,0 +1,129 @@
+
+// eslint-disable-next-line no-undef
+const list = document.getElementById('list');
+const addBook = document.getElementById('add-book');
+const contact = document.getElementById('contact-page');
+const booksContainer = document.createElement('ul');
+booksContainer.classList = 'book-list';
+const heading = document.querySelector('h1');
+const mainContainer = document.querySelector('main');
+
+let allBooks = [];
+if (localStorage.getItem('bookList') === null) {
+  localStorage.setItem('bookList', []);
+}
+if (localStorage.getItem('id') === null) {
+  localStorage.setItem('id', JSON.stringify(0));
+}
+
+// boik list
+const domDisplay = () => {
+  heading.innerHTML = 'All Awesome Books';
+  booksContainer.innerHTML = '';
+  mainContainer.innerHTML = '';
+  allBooks = JSON.parse(localStorage.getItem('bookList'));
+
+  allBooks.forEach((book) => {
+    const bookTitle = book.name;
+    const bookAuthor = book.author;
+    const bookId = book.id;
+    const removeBtn = document.createElement('button');
+    removeBtn.innerText = 'Remove';
+    removeBtn.addEventListener('click', (e) => {
+      const { id } = e.target.parentNode;
+      remove.remove(id);
+    });
+
+    const newBook = document.createElement('li');
+    const newTitle = document.createElement('p');
+
+    newTitle.innerHTML = `"${bookTitle}" by ${bookAuthor}`;
+    newBook.id = bookId;
+    newBook.appendChild(newTitle);
+    newBook.appendChild(removeBtn);
+    booksContainer.appendChild(newBook);
+    booksContainer.id = 'book-Container';
+    mainContainer.appendChild(booksContainer);
+  });
+};
+
+list.addEventListener('click', () => {
+  domDisplay();
+});
+
+// Add form page
+addBook.addEventListener('click', (e) => {
+  e.preventDefault();
+  heading.innerHTML = 'Add a New Book';
+  mainContainer.innerHTML = '';
+  const form = document.createElement('form');
+  form.classList += 'book-form';
+  //   titleInput
+  const titleInput = document.createElement('input');
+  titleInput.classList += 'form-control';
+  titleInput.type = 'text';
+  titleInput.name = 'bookname';
+  titleInput.id = 'title';
+  titleInput.placeholder = 'Title';
+  form.appendChild(titleInput);
+  //   authorInput
+  const authorInput = document.createElement('input');
+  authorInput.classList += 'form-control';
+  authorInput.type = 'text';
+  authorInput.name = 'authorname';
+  authorInput.id = 'author';
+  authorInput.placeholder = 'Author';
+  form.appendChild(authorInput);
+  //   addInput
+  const addBtn = document.createElement('button');
+  addBtn.classList += 'add-Btn';
+  addBtn.type = 'button';
+  addBtn.innerText = 'Add';
+  addBtn.id = 'btn';
+  form.appendChild(addBtn);
+  allBooks = JSON.parse(localStorage.getItem('bookList'));
+
+  mainContainer.appendChild(form);
+
+  addBtn.addEventListener('click', () => {
+    add.add(titleInput.value, authorInput.value);
+    titleInput.value = '';
+    authorInput.value = '';
+  });
+});
```

Comment:
- [ ] Please take a look at your `addForm` page. It is not rendering on my end so I'm unable to test the functionality, I'm getting the following errors:

![imageedit_4_4828126499](https://user-images.githubusercontent.com/77069011/214707426-b0d62211-505a-47d2-923e-b88c3167eb4a.gif)

This is happening because your code is not handling the case in which the "bookList" key in your local storage doesn't exist. Please consider providing it with a fallback value! 🤓 🔧